src/cgame/etj_jump_speeds.cpp

```diff
@@ -0,0 +1,202 @@
+
+namespace ETJump
+{
+	void DrawJumpSpeeds()
+	{
+		playerState_t* ps = getValidPlayerState();
+		float offX = 0.0f;
+		float offY = 0.0f;
+		float x1 = 6 + etj_jumpSpeedsX.value;
+		float x2 = 6 + 30 + etj_jumpSpeedsX.value;
+		float y1 = 240 + etj_jumpSpeedsY.value;
+		float y2 = 240 + 12 +etj_jumpSpeedsY.value;
+		bool drawShadow = etj_jumpSpeedsShadow.integer;
+		vec4_t color;
+		bool vertical = !etj_jumpSpeedsStyle.integer;
+
+		if (!etj_drawJumpSpeeds.integer)
+		{
+			return;
+		}
+
+		if (ps->persistant[PERS_TEAM] == TEAM_SPECTATOR)
+		{
+			return;
+		}
+
+		if (cg.zoomedBinoc || cg.zoomedScope)
+		{
+			return;
+		}
+
+		// draw the label text first
+		x1 = ETJump_AdjustPosition(x1);
+		x2 = ETJump_AdjustPosition(x2);
+		parseColorString(etj_jumpSpeedsColor.string, color);
+		DrawString(x1, y1, 0.2f, 0.2f, color, qfalse, "Jump Speeds:", 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+
+		// adjust x or y depending on style chosen
+		vertical ? y1 += 12 : x1 = DrawStringWidth("Jump Speeds: ", 0.2f) + 5;
+
+		for (auto i = 0; i < jumpSpeedHistory.size(); i++)
+		{
+			auto jumpSpeed = std::to_string(jumpSpeedHistory.at(i));
+			if (etj_jumpSpeedsShowDiff.integer)
+			{
+				AdjustColors(i, &color);
+			}
+			if (vertical)
+			{
+				// first column
+				if (i < 5)
+				{
+					DrawString(x1, y1, 0.2f, 0.2f, color, qfalse, jumpSpeed.c_str(), 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+					y1 += 12;
+				}
+				// second column
+				else
+				{
+					DrawString(x2, y2, 0.2f, 0.2f, color, qfalse, jumpSpeed.c_str(), 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+					y2 += 12;
+				}
+			}
+			// horizontal list
+			else
+			{
+				DrawString(x1, y1, 0.2f, 0.2f, color, qfalse, jumpSpeed.c_str(), 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+				x1 += 30;
+			}
+		}
+	}
+
+	void UpdateJumpSpeeds()
+	{
+		playerState_t* ps = getValidPlayerState();
+		// queue reset if last update was on different team
+		if (team != ps->persistant[PERS_TEAM])
+		{
+			resetQueued = true;
+		}
+		// if reset is queued, do that before we start storing new jump speeds
+		if (resetQueued)
+		{
+			ResetJumpSpeeds();
+			resetQueued = false;
+		}
+
+		team = ps->persistant[PERS_TEAM];
+		jumpSpeedHistory.push_back(ps->persistant[PERS_JUMP_SPEED]);
+		// we only want to keep last 10 jumps, so remove first value if we go over that
+		if (jumpSpeedHistory.size() > MAX_JUMPS)
+		{
+			// store the deleted speed so we can do diff comparisons later
+			lastDeletedSpeed = jumpSpeedHistory.front();
+			jumpSpeedHistory.erase(jumpSpeedHistory.begin());
+			jumpSpeedDeleted = true;
+		}
+	}
+
+	void QueueJumpSpeedsReset()
+	{
+		resetQueued = true;
+	}
+
+	void ResetJumpSpeeds()
+	{
+		jumpSpeedHistory.clear();
+		lastDeletedSpeed = 0;
+	}
+
+	void AdjustColors(int jumpNum, vec4_t* color)
+	{
+		auto alpha = (*color)[3];
+		vec4_t fasterColor = { 0.0, 1.0, 0.0, alpha };
```

Comment:
Is there a reason there is a cvar for jump speed in general, but not the diff?

Reply:
Didn't really think it would be necessary, but can implement ofc.